helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateComputationStage.java

```diff
@@ -113,6 +113,8 @@ public void process(ClusterEvent event) throws Exception {
       reportResourcePartitionCapacityMetrics(dataProvider.getAsyncTasksThreadPool(),
           clusterStatusMonitor, dataProvider.getResourceConfigMap().values());
 
+      // TODO: we only need to compute when there are resource using Waged. We should
+      // do this as perf improvement in future.
```

Comment:
Good callout. If we do this right, and call this only if the cluster has resources configured to use WAGED, then we should not even need to catch the exception in the other file (WagedInstanceCapacity). Is there an easy way to know if any resource is setup to use WAGED? 

(This comment doesn't block this PR, asking out of curiosity)

Reply:
This is correct. We don't need to compute so no need to catch the exception. 
The only way to know is by checking each resource config. We are doing this anyway in this stage so it shouldn't bring a lot extra cycles.